### PR TITLE
fix: watch tx confirmation event

### PR DESCRIPTION
**Pull request by UncleBill**
https://mask.atlassian.net/browse/MF-299

### CODE REVIEW

**Review by Jack-Works — DISMISSED** [review]

**Review by hanszhoou — DISMISSED** [review]

File: packages/mask/src/plugins/Trader/trader/bancor/useTradeCallback.ts [hunk 1]
```diff
@@ -59,21 +66,29 @@ export function useTradeCallback(tradeComputed: TradeComputed<SwapBancorRequest>
...
+            web3.eth
+                .sendTransaction(config_, (error, hash) => {
+                    if (error) {
+                        setTradeState({
+                            type: TransactionStateType.FAILED,
+                            error,
+                        })
+                        reject(error)
+                    } else {
+                        setTradeState({
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Trader/trader/bancor/useTradeCallback.ts]: Don't push tx state both. Plz remove hash.

Comment (UncleBill) [reply packages/mask/src/plugins/Trader/trader/bancor/useTradeCallback.ts]: Fixed

File: packages/mask/src/plugins/Trader/trader/0x/useTradeCallback.ts [hunk 2]
```diff
@@ -58,21 +65,29 @@ export function useTradeCallback(tradeComputed: TradeComputed<SwapQuoteResponse>
...
+            web3.eth
+                .sendTransaction(config_, (error, hash) => {
+                    if (error) {
+                        setTradeState({
+                            type: TransactionStateType.FAILED,
+                            error,
+                        })
+                        reject(error)
+                    } else {
+                        setTradeState({
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Trader/trader/0x/useTradeCallback.ts]: Same here.

File: packages/mask/src/plugins/Trader/trader/dodo/useTradeCallback.ts [hunk 3]
```diff
@@ -56,21 +63,29 @@ export function useTradeCallback(
...
+            web3.eth
+                .sendTransaction(config_, (error, hash) => {
+                    if (error) {
+                        setTradeState({
+                            type: TransactionStateType.FAILED,
+                            error,
+                        })
+                        reject(error)
+                    } else {
+                        setTradeState({
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Trader/trader/dodo/useTradeCallback.ts]: Don't push tx state both. Plz remove the hash event listener.

File: packages/mask/src/plugins/Trader/trader/uniswap/useTradeCallback.ts [hunk 4]
```diff
@@ -135,43 +142,51 @@ export function useTradeCallback(
...
+                                })
+                                reject(error_)
+                            } else {
+                                setTradeState({
+                                    type: TransactionStateType.FAILED,
+                                    error: new Error(`Swap failed: ${swapErrorToUserReadableMessage(error)}`),
+                                })
+                            }
                         } else {
                             setTradeState({
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Trader/trader/uniswap/useTradeCallback.ts]: Don't push tx state both. Plz remove the hash event listener.

File: packages/mask/src/plugins/Trader/trader/openocean/useTradeCallback.ts [hunk 5]
```diff
@@ -61,21 +62,29 @@ export function useTradeCallback(
...
+            web3.eth
+                .sendTransaction(config_, (error, hash) => {
+                    if (error) {
+                        setTradeState({
+                            type: TransactionStateType.FAILED,
+                            error,
+                        })
+                        reject(error)
+                    } else {
+                        setTradeState({
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Trader/trader/openocean/useTradeCallback.ts]: Don't push tx state both. Plz remove the hash event listener.

**Review by UncleBill — COMMENTED** [review]
@DimensionDev/qa Please help to verify.

**Review by guanbinrui — DISMISSED** [review]

**Comment by guanbinrui** [comment]
@DimensionDev/qa

**Comment by wenluomask** [comment]
Please check comments in Jira.

File: packages/web3-shared/evm/hooks/useTransactionCallback.ts [hunk 6]
```diff
@@ -44,14 +44,6 @@ export function useTransactionCallback<T extends unknown>(
         return new Promise<void>(async (resolve, reject) => {
             method
                 .send(gasExpectedConfig)
-                .once(TransactionEventType.TRANSACTION_HASH, (hash) => {
```
Comment (guanbinrui) [inline packages/web3-shared/evm/hooks/useTransactionCallback.ts:47]: Why not also remove the `TransactionEventType.RECEIPT` event?

**Comment by guanbinrui** [comment]
Please resolve the conflicts.

**Comment by UncleBill** [comment]
Succeeded by https://github.com/DimensionDev/Maskbook/pull/6181

**Pull request closed without merging**